packages/devtools_app/lib/src/screens/profiler/panes/method_table/method_table_model.dart

```diff
@@ -27,48 +37,102 @@ class MethodTableGraphNode extends GraphNode {
+  final ProfileMetaData profileMetaData;
 
-  // TODO(kenz): implement the calculation for exclusive and inclusive count.
+  String get id => '$name-$_sourceUri';
 
-  /// The number of cpu samples where this frame is on top of the stack.
-  ///
-  /// This count is used to calculate self time.
-  int exclusiveSampleCount = 0;
+  String get display =>
+      '$name${_sourceUri.isNotEmpty ? ' - ($_sourceUri)' : ''}';
 
-  /// The number of cpu samples where this frame is anywhere on the stack.
-  ///
-  /// This count is used to calculate total time.
-  int inclusiveSampleCount = 0;
+  /// The number of cpu samples where this method is on top of the stack.
+  int get selfCount => _selfCount;
+  late int _selfCount;
 
-  late double totalTimeRatio =
-      safeDivide(inclusiveSampleCount, profileMetaData.sampleCount);
+  /// The number of cpu samples where this method is anywhere on the stack.
+  int get totalCount => _totalCount;
+  late int _totalCount;
 
-  late Duration totalTime = Duration(
-    microseconds:
-        (totalTimeRatio * profileMetaData.time!.duration.inMicroseconds)
-            .round(),
-  );
+  double get selfTimeRatio =>
+      safeDivide(selfCount, profileMetaData.sampleCount);
 
-  late double selfTimeRatio =
-      safeDivide(exclusiveSampleCount, profileMetaData.sampleCount);
+  Duration get selfTime => Duration(
+        microseconds:
+            (selfTimeRatio * profileMetaData.time!.duration.inMicroseconds)
+                .round(),
+      );
 
-  late Duration selfTime = Duration(
-    microseconds:
-        (selfTimeRatio * profileMetaData.time!.duration.inMicroseconds).round(),
-  );
+  double get totalTimeRatio =>
+      safeDivide(totalCount, profileMetaData.sampleCount);
 
-  double get inclusiveSampleRatio => safeDivide(
-        inclusiveSampleCount,
-        profileMetaData.sampleCount,
+  Duration get totalTime => Duration(
+        microseconds:
+            (totalTimeRatio * profileMetaData.time!.duration.inMicroseconds)
+                .round(),
       );
 
-  double get exclusiveSampleRatio => safeDivide(
-        exclusiveSampleCount,
-        profileMetaData.sampleCount,
-      );
+  void merge(MethodTableGraphNode other) {
+    if (!shallowEquals(other)) return;
+    _totalCount += other.totalCount;
+    _selfCount += other.selfCount;
+  }
+
+  bool shallowEquals(other) {
+    return other is MethodTableGraphNode &&
+        other.name == name &&
+        other._sourceUri == _sourceUri;
+  }
+
+  @override
+  String toString() {
+    String generateDisplayFor(
+      Set<GraphNode> nodes, {
+      required double Function(MethodTableGraphNode) percentCallback,
+    }) {
+      return nodes
+          .cast<MethodTableGraphNode>()
+          // Sort in descending order.
+          .sorted((a, b) => percentCallback(b).compareTo(percentCallback(a)))
+          .map(
+            (node) => '${node.display} - ${percent2(percentCallback(node))}',
+          )
+          .join('\n    ');
```

Comment:
What's with the spaces after the newlines?

Reply:
It is for an indent in the display

Reply:
will make a named variable to make that more clear